unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,176 @@
+        check_transaction_id_cached_out_call_success_fixture._check_transaction_id_cached_out.assert_called_once()
+
+    def test_check_buffer_message_matching_command_and_id_throws_exception(self):
+        message = {}
+        with pytest.raises(RedisAPIException,
+                           match='The answer message does not contain a valid '
+                                 '\'transaction_id\' member.') as ex:
+            redis_client_base = RedisClientBase(area_id=area_id, autoregister=False)
+            redis_client_base._check_buffer_message_matching_command_and_id(message)
+
+    def test_register_blocking_true_throws_exception(self):
+        with pytest.raises(RedisAPIException,
+                           match='API registration process timed out. Server will continue processing '
+                                 'your request on the background and will notify you as soon as the '
+                                 'registration has been completed.') as ex:
+            redis_client_base = RedisClientBase(area_id=area_id, autoregister=True)
+            redis_client_base.register(is_blocking=True)
+
+    def test_register_blocking_false_published_success(self):
+        # arrange
+        redis_client_base = RedisClientBase(area_id=area_id, autoregister=False)
+        redis_client_base.register(is_blocking=False)
+        assert redis_client_base._blocking_command_responses["register"]["name"] == area_id
+
+    def test_register_blocking_false_throw_exception(self, redis_client_auto_register):
+        with pytest.raises(RedisAPIException,
+                           match='API is already registered to the market.') as ex:
+            self.redis_client_base = redis_client_auto_register
+            self.redis_client_base.is_active = True
+            self.redis_client_base.register(is_blocking=False)
+
+    def test_unregister_blocking_isactive_false_throws_exception(self, redis_client_auto_register):
+        with pytest.raises(RedisAPIException,
+                           match='API is already unregistered from the market.') as ex:
+            self.obj = redis_client_auto_register
+            self.obj.is_active = False
+            self.obj.unregister(is_blocking=True)
+
+    def test_unregister_blocking_isactive_true_throws_exception(self, redis_client_auto_register):
+        with pytest.raises(RedisAPIException,
+                           match='API unregister process timed out. Server will continue processing '
+                                 'your request on the background and will notify you as soon as '
+                                 'the unregistration has been completed.') as ex:
+            self.obj = redis_client_auto_register
+            self.obj.is_active = True
+            self.obj.unregister(is_blocking=True)
+
+    def test_select_aggregator_self_area_uuid_none_throws_exception(self, redis_client_auto_register):
```

Comment:
Here you are only testing the corner cases when there is no connection to the d3a.
This is good, but you also need to test the ideal path.
As I mentioned in the offline discussion, here again an example:
If (after) the `select_aggregator(is_blocking=False)` function is called, the following action should be done, conditions should be True:
1. `self._transaction_id_buffer` should not be empty
2. `self.redis_db.publish` should have been called with the expected `data` JSON string.